packages/next/src/client/components/router-reducer/reducers/server-action-reducer.ts

```diff
@@ -71,7 +75,9 @@ async function fetchServerAction(
   { actionId, actionArgs }: ServerActionAction
 ): Promise<FetchServerActionResult> {
   const temporaryReferences = createTemporaryReferenceSet()
-  const body = await encodeReply(actionArgs, { temporaryReferences })
+  const info = extractInfoFromServerReferenceId(actionId)
+  const filteredArgs = filterArgs(actionArgs, info)
```

Comment:
hmmm i wonder if there's a better verb. `filter` kinda suggests that it acts like `Array#filter` (which'd obviously be bad). maybe something like `maskUnusedArgs`? "mask" as in https://en.wikipedia.org/wiki/Layers_(digital_image_editing)#Layer_mask

Reply:
or `removeUnusedArgs`?

Reply:
settled on `omitUnusedArgs`